cc/tf_bt_utils.cc

```diff
@@ -82,5 +91,92 @@ void WriteGameExamples(const std::string& gcp_project_name,
+    if (cell.family_name() == "metadata" &&
+        cell.column_qualifier() == "game_counter") {
+      return cell.value_as<bigendian64_t>().get();
+    }
+  }
+  MG_FATAL() << "Failed to increment table_state=metadata:game_counter";
+  return 0;
+}
+
+void PortGamesToBigtable(const std::string& gcp_project_name,
+                         const std::string& instance_name,
+                         const std::string& table_name,
+                         const std::vector<std::string>& paths,
+                         int64_t game_counter) {
+  auto client_options = google::cloud::bigtable::ClientOptions();
+  auto channel_args = client_options.channel_arguments();
+  channel_args.SetUserAgentPrefix("minigo_to_cbt");
+  channel_args.SetInt(GRPC_ARG_ENABLE_CENSUS, 0);
+  channel_args.SetInt(GRPC_ARG_KEEPALIVE_PERMIT_WITHOUT_CALLS, 0);
+  channel_args.SetInt(GRPC_ARG_KEEPALIVE_TIMEOUT_MS, 60 * 1000);
+  client_options.set_channel_arguments(channel_args);
+  Table table(
+      CreateDefaultDataClient(gcp_project_name, instance_name, client_options),
+      table_name);
+
+  if (game_counter < 0) {
+    if (paths.size() != 1) {
+      MG_FATAL() << "Atomic game updates require batch size of 1 game";
+      return;
+    }
+    MG_FATAL() << "Have not yet implemented atomic game counter update.";
+    return;
+  }
+
+  google::cloud::bigtable::BulkMutation game_batch;
+  auto start_time = absl::Now();
+  int changes = 0;
+  for (const auto& path : paths) {
+    std::unique_ptr<tensorflow::RandomAccessFile> file;
+    TF_CHECK_OK(tensorflow::Env::Default()->NewRandomAccessFile(path, &file));
+
+    RecordReaderOptions options;
+    options.compression_type = RecordReaderOptions::ZLIB_COMPRESSION;
+    SequentialRecordReader reader(file.get(), options);
+
+    auto row_prefix = absl::StrFormat("g_%010d", game_counter);
+
+    auto game_id = path;
+    game_id.erase(game_id.rfind(".tfrecord.zz"));
+    game_id[game_id.rfind('/')] = '-';
+    game_id.erase(0, game_id.rfind('/') + 1);
+    auto zero_row = absl::StrFormat("%s_m_%06d", row_prefix, 0);
+    google::cloud::bigtable::SingleRowMutation zero_row_mutation(zero_row);
+    zero_row_mutation.emplace_back(SetCell("metadata", "game_id", game_id));
+    game_batch.emplace_back(std::move(zero_row_mutation));
+
+    std::string data;
+    int move_number = 0;
+    while (reader.ReadRecord(&data).ok()) {
+      auto row_name = absl::StrFormat("%s_m_%06d", row_prefix, move_number);
+      google::cloud::bigtable::SingleRowMutation row_mutation(row_name);
+      row_mutation.emplace_back(SetCell("tfexample", "example", data));
+      row_mutation.emplace_back(
+          SetCell("metadata", "move", std::to_string(move_number)));
+      game_batch.emplace_back(std::move(row_mutation));
+      move_number++;
+      changes++;
+    }
+    game_counter++;
+  }
+
+  table.BulkApply(std::move(game_batch));
+  auto finish_time = absl::Now();
+  double elapsed = absl::ToDoubleSeconds(finish_time - start_time);
```

Comment:
seems unused

Reply:
Sure enough. I had been having this routine log the individual batch times, but once I was finished, only the top-level stats were interesting. I'll add the report using `VLOG(2)` (from tensorflow), unless there is a reason to use something else.